frontend/src/app/modules/common/form-field/form-field.component.ts

```diff
@@ -16,32 +18,38 @@ import {
 export class OpFormFieldComponent {
   @HostBinding('class.op-form-field') className = true;
   @HostBinding('class.op-form-field_invalid') get errorClassName() {
-    return this.isInvalid;
+    return this.showErrorMessage;
   }
 
   @Input() label = '';
+  @Input() inlineLabel = true;
   @Input() required = false;
+  @Input() showValidationErrorOn: 'change' | 'blur' | 'submit' | 'never' = 'submit';
```

Comment:
This should be the same behaviour for all fields imo, and even if we could be setting this per-field, then it should be set on the model, not on the component.  I'm not sure why you changed the `isDirty` check, did it not work at all?

Reply:
In any case, removing the current behaviour for the `isDirty` check breaks the validation behaviour inside the ium: https://github.com/opf/openproject/blob/dev/frontend/src/app/modules/invite-user-modal/project-selection/project-selection.component.ts#L93

Reply:
The issue here was that, whenever a parent field was dirty (whenever any child field was dirty), all the required child fields were showing the validation error even when they were not dirty. It was also not possible to customize when the validation errors should be shown (on 'submit' in this case).

In this implementation, I just wanted to allow this customization. In the ium case, probably showValidationErrorOn === 'change' could work (this.formControl.invalid && this.formControl.dirty). I thought about allowing any expression in the input (e.g. @showValidationErrorOn="formControl.invalid && whatever") but finally decided to keep it tied to the Angular forms logic.

Would it work for you?